main.py

```diff
@@ -1,13 +1,102 @@
 #!/usr/bin/env python
+import re
+from nltk.wsd import lesk
 
-# Your classes or methods go here:
 
+class JsonParsedWriter:
 
-class Tool:
-    def my_method(self):
-        print("my_method called!")
+    @staticmethod
+    def write(data, output_name, indent=2):
+        import json
+        with open(output_name, 'w') as out_file:
+            parsed = json.loads(json.dumps(data))
+            json.dump(parsed, out_file, indent=indent)
+
+
+class TweetCleaner:
+
+    def __init__(self, input_file_name):
+
+        self.name = input_file_name
+
+        # list of patterns for regular expressions
+        self.tags_patt = re.compile(r",\[.*\]")
+        self.clr_tags_patt = re.compile(r"\w+")
+        self.body_patt = re.compile(r"#\w+|@\w+")
+        self.clr_body_patt = re.compile(r"\w+")
+        self.url_patt = re.compile(r"https?://[A-Za-z0-9./]+")
+        self.sign_patt = re.compile(r"$\w+")
+        self.token_skip_patt = re.compile(r"#\w+|@\w+|\$\w+|https?://[A-Za-z0-9./]+|")
+        self.token_clr_patt = re.compile(r"[A-Za-z0-9.\-:]+")
+
+    def __clr_tags_to_metadata(self, row):
+        # task 1 and 2  for the tags array || and task 3 for the text
+        self.sign_result = re.findall(self.sign_patt, row)
+        self.tags_result = re.findall(self.tags_patt, row)
+        tags_clr = re.findall(self.clr_tags_patt, str(self.tags_result))
+        url_result = re.findall(self.url_patt, row)
+        self.metadata = tags_clr + url_result
+
+    def __remove_dollar_sign_words(self, row, tag_results):
+        # task 1 for the text and processing text
+        self.body = str()
+        for tag in tag_results:
+            self.body = row.replace(tag, '')
+
+        self.clr_row = self.body.replace("&amp;", "&", ) \
+            .replace("&#39;", "'").replace("&quot;", '"').replace('&rsquo;', '\'')
+
+        for sign in self.sign_result:
+            self.clr_row = self.clr_row.replace(sign, '')
+
+    def __place_tags_wrd_to_body_tags(self, clr_row):
+        # task 2 for the text
+        self.body_tags = re.findall(self.body_patt, clr_row)
```

Comment:
Writing into the internal state - this should be avoided. Make it the return value of the method.